proposals/oci-support-updates.md

```diff
@@ -0,0 +1,95 @@
+
+As of yet, the code related to registry support has been written standalone (see `internal/experimental/registry`).
+
+The act of downloading a chart from an OCI registry should mimic what is already possible using downloader plugins. In theory, using a protocol prefix such as `oci://` in any place where chart repos are referenced should work just by implementing a `Getter` for OCI.
+
+Currently there is no Helm-specific way to upload chart packages. Using the same model of `Getter`, a new interface called `Pusher` should be introduced, with a single built-in implementation (OCI). This opens the door for plugins to implement their own upload mechanism, and for Helm to add a new top-level upload command such as `helm push`.
+
+
+## 2. Support for provenance files
+
+Helm currently has the ability to verify package signatures, assuming the presence of a file suffixed with `.prov` sitting next to a chart `.tgz` in a repository.
+
+Support for this method of signature validation should be carried over into OCI storage. As the format of the OCI manifest is custom to Helm, Helm can also choose to modify the resulting manifest on upload when a chart is being signed (e.g. `helm push --sign`).
+
+As far as the low-level details, the `.prov` file will simply be stored as a second layer on the manifest. If running `helm pull --verify oci://...`, the second layer (layer at index 1) will be assumed to be the provenance file. The first layer is always the chart `.tgz` itself.
+
+Here is an example of what a manifest might look like with a provenance file attached:
+
+```
+{
+  "schemaVersion": 2,
+  "config": {
+    "mediaType": "application/vnd.cncf.helm.config.v1+json",
+    "digest": "sha256:8ec7c0f2f6860037c19b54c3cfbab48d9b4b21b485a93d87b64690fdb68c2111",
+    "size": 117
+  },
+  "layers": [
+    {
+      "mediaType": "application/tar+gzip",
+      "digest": "sha256:1b251d38cfe948dfc0a5745b7af5ca574ecb61e52aed10b19039db39af6e1617",
+      "size": 2487
+    },
+    {
+      "mediaType": "application/pgp-signature",
+      "digest": "sha256:3e207b409db364b595ba862cdc12be96dcdad8e36c59a03b7b3b61c946a5741a",
+      "size": 643
+    }
+  ]
+}
+
+```
+
+If `application/pgp-signature` is not appropriate based the contents of a `.prov` file (e.g. it does not meet the IANA requirements), a custom media type will be used such as `application/vnd.cncf.helm.provenance.v1`.
+
+## 3. Chart versions == OCI reference tags
+
+To keep things simple, the version of a chart will be 1-to-1 with the tag used on registry references. Arbitrary tags will not be supported for initial release. As time goes on, support for arbitrary tags may be reintroduced.
+
+This also means that tags are no longer necessary to be provided on the command-line in the form `<ref>:<tag>`. Some users feel strongly to keep this to match the UX of Docker and other tools, so perhaps this feature is kept and treated the same as using the `--version` flag. Thoughts?
```

Comment:
I like the idea of `:<tag>` being the same as the `--version` flag. I've not thought much about it, though. We would need to verify that the `--version` flag and tag are not in conflict with each other.

Reply:
Perhaps we just throw an error if both are provided. I think the `:<tag>` syntax is useful, allowing users to copy and paste a single string